real_time/surface_src/java_src/Alice/src/main/java/com/roboclub/robobuggy/simulation/SensorPlayer.java

```diff
@@ -225,4 +270,11 @@ private boolean validateLogFileMetadata(JsonObject logFile) {
 
         return true;
     }
+
+    /**
+     * gets the new playback speed from the GUI and puts it into playbackSpeed
+     */
+    public void getNewPlaybackSpeed() {
+        playbackSpeed = Gui.getInstance().getMainGuiWindow().getCtrlPanel().getLoggingPanel().getPlaybackSpeed();
+    }
```

Comment:
this is super ugly can you change it to just making the loggingPanel a factory so you can directly reference it?